Source/WebCore/loader/ProgressTracker.cpp

```diff
@@ -149,27 +151,27 @@ void ProgressTracker::progressEstimateChanged(LocalFrame& frame)
 
 void ProgressTracker::progressCompleted(LocalFrame& frame)
 {
-    LOG(Progress, "Progress completed (%p) - frame %p(frameID %" PRIu64 "), value %f, tracked frames %d, originating frame %p", this, &frame, frame.frameID().object().toUInt64(), m_progressValue, m_numProgressTrackedFrames, m_originatingProgressFrame.get());
-    PROGRESS_TRACKER_RELEASE_LOG(PROGRESSTRACKER_PROGRESSCOMPLETED, frame.frameID().object().toUInt64(), m_progressValue, m_numProgressTrackedFrames, m_originatingProgressFrame ? m_originatingProgressFrame->frameID().object().toUInt64() : 0, m_isMainLoad);
+    RefPtr originatingProgressFrame = m_originatingProgressFrame.get();
+    LOG(Progress, "Progress completed (%p) - frame %p(frameID %" PRIu64 "), value %f, tracked frames %d, originating frame %p", this, &frame, frame.frameID().object().toUInt64(), m_progressValue, m_numProgressTrackedFrames, originatingProgressFrame.get());
+    PROGRESS_TRACKER_RELEASE_LOG(PROGRESSTRACKER_PROGRESSCOMPLETED, frame.frameID().object().toUInt64(), m_progressValue, m_numProgressTrackedFrames, originatingProgressFrame ? originatingProgressFrame->frameID().object().toUInt64() : 0, m_isMainLoad);
```

Comment:
Doesn’t matter either way, but smart pointer guidelines say we don’t need smart pointers on the stack when objects only call trivial and inlined functions, which is the case here.